src/Analyzers/MSTest.Analyzers/TestInitializeShouldBeValidAnalyzer.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Immutable;
+
+using Analyzer.Utilities.Extensions;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+using MSTest.Analyzers.Helpers;
+
+namespace MSTest.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class TestInitializeShouldBeValidAnalyzer : DiagnosticAnalyzer
+{
+    private static readonly LocalizableResourceString Title = new(nameof(Resources.TestInitializeShouldBeValidTitle), Resources.ResourceManager, typeof(Resources));
+    private static readonly LocalizableResourceString Description = new(nameof(Resources.TestInitializeShouldBeValidDescription), Resources.ResourceManager, typeof(Resources));
+    private static readonly LocalizableResourceString MessageFormat = new(nameof(Resources.TestInitializeShouldBeValidMessageFormat_Public), Resources.ResourceManager, typeof(Resources));
+
+    internal static readonly DiagnosticDescriptor PublicRule = DiagnosticDescriptorHelper.Create(
```

Comment:
Could you check if when we set `[assembly: DiscoverInternals]` the `TestInitialize` allows internal? I would expect the behavior to be similar to `TestMethod` attribute. If that's the case, you can mimic the logic from the `TestMethodShouldBeValidAnalyzer` class.

Reply:
it does not allow internal even with [assembly: DiscoverInternals]

Reply:
I really love this discover internals that only work halfway and doesn't allow you to have it all internal xD

Reply:
yeah i tested that and it's not working with internal